File: backend/app/services/juniper.py
from netmiko import ConnectHandler


def fetch_juniper_config(device) -> str:
    conn = ConnectHandler(
        device_type="juniper_junos",
        host=device.ip_address,
        port=device.port,
        username=device.ssh_username,
        password=device.ssh_password,
        timeout=30,
    )
    output = conn.send_command("show configuration | display set", read_timeout=60)
    conn.disconnect()
    return output


def test_juniper(device):
    conn = ConnectHandler(
        device_type="juniper_junos",
        host=device.ip_address,
        port=device.port,
        username=device.ssh_username,
        password=device.ssh_password,
        timeout=10,
    )
    output = conn.send_command("show version", read_timeout=15)
    conn.disconnect()
    return output

File: backend/internal/vendor_client/juniper.go
package vendor_client

import (
	"fmt"
	"time"
)

func FetchJuniperConfig(ip string, port int, username, password string) (string, error) {
	commands := []string{"show configuration | display set | no-more"}
	output, err := runSSHCommands(ip, port, username, password, commands, 30*time.Second, 60*time.Second)
	if err != nil {
		return "", fmt.Errorf("SSH connection failed: %v", err)
	}
	cleaned := cleanSSHOutput(output, "show configuration")
	return cleaned, nil
}

func TestJuniper(ip string, port int, username, password string) error {
	commands := []string{"show version"}
	_, err := runSSHCommands(ip, port, username, password, commands, 10*time.Second, 15*time.Second)
	if err != nil {
		return fmt.Errorf("SSH connection failed: %v", err)
	}
	return nil
}
